# Reanalysis the BRCA data

The RAW bam data download from GDC is extracted to subbam for recheck. And the calling process was done through GATK. Ignoring the `hard filter` provided by the GATK. Now I walk through the whole process for the project.

The outline follows the instruction:  
preprocessing:

1. Download BRCA WXS data from [GDC]('https://gdc-portal.nci.nih.gov/projects/TCGA-BRCA').
2. Dwonload Regulatory Build from Ensembl (release-v86).
3. Call mutation at regulatory regions by GATK (**-L regulatoryRegion.bed**)

Filteration: 

1. R script for obtain somatic mutation for every sample.
  + `02.getSomatic.R`
  + manifest statistics for file size and mutation.
  + Normal: MQ > 20, DP >= 5, altDP >= 3; write to 02.filtersnp
  + Tumor: MQ > 20, DP >= 10, altDP >= 3; write to 02.filtersnp
  + Somatic: $Tumor -  Normal for one case$; write to 03.somatic/01.annotation
2. Bash script ANNOVAR to filter dbSNP
  + `removedbSNP.sh`
3. Annotate with regulatory region.
  + `regulatoryFeture.py`
4. Transform file to rds for analysis
  + `03.loadAnnoSomatic.R`
5. Reduce false positive site
  + `04.somaticMutationAnalysis.R`
  + normalMutation recurrent >=1
  + truePositiveSomaticMutation = somaticMutation - normalMutation
  + draw **01.SomaticMutationStat.png**
  + Filter position which recurrent >= 5
6. Refined recurrent mutation.
  + 60 mutation positions recur >=6 and save them to scan all mpileups.
  + `extractCoverage.sh` extract position from all samples. 
  + The process is working for the ratio and ensure the position is a real somatic mutation.
7. Use biomaRt to get target genes for mutation
  + `05.targetGeneForMutation.R` get nearest target genes.
  + mark as Oncogene or tumor supressor genes.
8. Target gene expression
  + `06.targetGeneExpression.R` calculate gene expression.
9. Recheck the mutation point.
  + Get mpileup coverage `extractCoverage.sh`
  + Load point mpileup file for recheck, the position cov don't attached with base quality.
  + 

 
 
 